dired-open-with.el

```diff
@@ -108,7 +108,8 @@ Every application is represented as a Hash Table."
   "Format XDG application EXEC string with PATH and return an executable command.
 For the list of keys and their meaning, please see
 https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s07.html"
-  (let* ((url path)
+  (let* ((path (shell-quote-argument path))
+	 (url path)
```

Comment:
Sorry for nitpicking but there is a TAB character instead of spaces at the beginning of this line.

![Screenshot_2024-06-04_18-13-40](https://github.com/FrostyX/dired-open-with/assets/2151771/9450053e-920b-4f09-8ff4-06d6ca70a2e4)

Reply:
The `shell-quote-argument` behaves strange though.

Why does `(shell-quote-argument "image with space.png")` evaluate to `"image\\ with\\ space.png"`? The backslashes shouldn't be doubled.

Reply:
Ah, I finally figured out the problem. We must do:

```
(let* ((path (shell-quote-argument (file-truename path)))
```

With this, I can finally open files with spaces

Reply:
> The `shell-quote-argument` behaves strange though.
> 
> Why does `(shell-quote-argument "image with space.png")` evaluate to `"image\\ with\\ space.png"`? The backslashes shouldn't be doubled.

It's just the elisp [printed representation](https://www.gnu.org/software/emacs/manual/html_node/elisp/Printed-Representation.html) of a backslash.

Reply:
> Ah, I finally figured out the problem. We must do:
> 
> ```emacs-lisp
> (let* ((path (shell-quote-argument (file-truename path)))
> ```
> 
> With this, I can finally open files with spaces

I could not reproduce the problem, but in fact the parameter should be an absolute path [according to XDG spec](https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s07.html) ). I have changed the commit again (without tabs).